client/packages/programs/src/JsonForms/common/components/Select.tsx

```diff
@@ -168,22 +185,69 @@ const getHighlightParts = (
   );
 };
 
+/** Either the full list or the narrowed list from conditional option */
+
+const useFilteredItems = (
+  items: string[] | undefined,
+  current: string | undefined,
+  options: Options | undefined
+): [string[], string | undefined] => {
+  const { core } = useJsonForms();
+  const [error, setError] = useState<string | undefined>();
+  const [visibleItems, setVisibleItems] = useState(items ?? []);
+
+  const conditionField = extractProperty(
+    core?.data ?? {},
+    options?.fieldFilter?.field ?? ''
+  );
+  useEffect(() => {
+    if (!items || !options?.fieldFilter) {
+      return;
+    }
+    const mapping = options.fieldFilter.mapping[conditionField] ?? [];
+    const filtered = items.filter(item => mapping.includes(item));
+    if (current !== undefined && !filtered.includes(current)) {
+      setVisibleItems([current, ...filtered]);
+      setError('Please select a valid option');
+    } else {
+      setVisibleItems(filtered);
+      setError(undefined);
+    }
+  }, [options, current, conditionField]);
+
+  useEffect(() => {}, [current, visibleItems]);
```

Comment:
` useEffect(() => {}, [current, visibleItems]);`
Assume this shouldn't be here?